### DISCUSSION TITLE
Logging System

### DISCUSSION THREAD

**Discussion by thebonbon** [EveronLife/EveronLife#74] [Ideas]
[@Kasper](https://github.com/EveronLife/EveronLife/pulls/KasperOnPC) and I are working on a Logging System.
This could be part of **P2 - Advanced farming**

https://user-images.githubusercontent.com/45827986/171670139-570ed53e-8369-415e-ac21-58c5ae4ee9c7.mp4

**Plans:**
- Bring logs to saw mill and process them to planks that you can then sell (or use to build furniture etc. But thats for another project)
- Area in which tree can be chopped. 
  -> Problem: Cant add Components via script. 
  -> Possible solution: Spawn custom trees in that area (random / grid)
- No area, all trees can be chopped? (+ blacklist / whitelist area?) -> Need to override every tree..

Any ideas, suggestions or comments?

**Comment by Arkensor** [comment]
It looks super cool. The problem of how to decide what trees are usable for the system and which are not is the reason we did not plan anything regarding logging so far. Enabling this for all trees (technical aspects aside) would make it nearly impossible to balance wood cutting. Anything near a sawmill would be chopped down. But only enabling trees in a certain region is somewhat emersion-breaking.

One thing that comes to my mind, is that maybe we could randomly select a tree every X time from the world and some ❌ marker onto it visually in yellow or something. That way lore-wise those are rotting trees that need to be cut down or something. And cutting any other trees does not work. This way the job could become a unique one in that you need to go around the map and look for those that are marked to be choppable. 

Interesting feature. We will just need to figure out a way to control it.

**Reply by thebonbon** [reply]
> Enabling this for all trees (technical aspects aside) would make it nearly impossible to balance wood cutting. Anything near a sawmill would be chopped down.
>
Could also lead to some interesting gameplay. You have to go further and further to get wood.
-> What about a blacklisted area around buildings / cites  etc.. ? So you have to go further away to get wood

> One thing that comes to my mind, is that maybe we could randomly select a tree every X time from the world and some ❌ marker onto it visually in yellow or something. That way lore-wise those are rotting trees that need to be cut down or something. And cutting any other trees does not work. This way the job could become a unique one in that you need to go around the map and look for those that are marked to be choppable.
>
Would be hard to see in a big forest. You would spend hours running around in a forest to find the X on a tree.

**Reply by Arkensor** [reply]
> Would be hard to see in a big forest. You would spend hours running around in a forest to find the X on a tree.

Exactly, that's the point. We could have server owners place down a zone prefab that marks some trees as choppable and they could be placed along streets or something. So to avoid them falling onto people they need to be cut down for "safety".

**Reply by jasper-one** [reply]
For the "control" part, what about simply having areas marked on the map, where all the trees can be chopped down in a certain zone? Kind of like a plantation in real life where you can go (in the future maybe you purchase a permit to be able to log/chop there).

There are actually some great areas on the map as well that have trees planted in a plantation-style like here: 

![image](https://user-images.githubusercontent.com/106525617/171958971-26f5b107-2eb4-4cc3-9555-0b07dfd7980a.png)

I think this would be great for immersion and gameplay because you know that all trees in the area can be cut, it will also group players into certain areas to ensure interactions.

**Reply by Arkensor** [reply]
> I think this would be great for immersion and gameplay because you know that all trees in the area can be cut, it will also group players into certain areas to ensure interactions.

I think you might be right. For RP purposes it might make sense to have a forest area that respawns some of the trees after X time again so people are likely to meet others there that are cutting wood.

**Reply by jasper-one** [reply]
I'm glad you agree. I think that, that would mot only apply here but also to other gathering in the future. Such as mining certian materials, harvesting certian plants (e.g drug fields) as you can find assets for most things across the map but it will be important you can't just do every activity anywhere.

**Comment by hiimk3v** [comment]
Incredible!

**Comment by thebonbon** [comment]
I added an area that enables damage on every tree inside:
![loggingArea](https://user-images.githubusercontent.com/45827986/172307658-9edcb292-1e66-4b02-9f4a-ce496355739d.png)


I also added collision damage:

https://user-images.githubusercontent.com/45827986/172307866-d5fa2dc4-39d2-4555-933a-19ac8cbf2e15.mp4

**Reply by LuxPoseidon** [reply]
they may need some extra "health" or resistance againced anything not an axe, you know, damage a vehicle when it hits it, explosions should probably have less effect.

But good demonstration, I like it.

**Comment by thebonbon** [comment]
Waiting for https://github.com/EveronLife/EveronLife/pull/88 to be merged, so I can add the axe and make a PR

**Reply by Arkensor** [reply]
Will review that later today
